mllib/src/main/scala/org/apache/spark/ml/feature/Word2Vec.scala

```diff
@@ -227,7 +227,7 @@ class Word2VecModel private[ml] (
    */
   @Since("1.5.0")
   def findSynonyms(word: String, num: Int): DataFrame = {
-    findSynonyms(wordVectors.transform(word), num)
+    findSynonyms(wordVectors.transform(word), num, Some(word))
```

Comment:
I see. I agree that something gets a little bit duplicated no matter what. Given your change, it seems easiest to pass the string vs vector argument all the way down, even if that means in these other two classes you duplicate a little code to transform the dataframe, etc. If it's nontrivial, sure, a little helper method seems reasonable. That would help keep this layered more cleanly IMHO.